src/Compute/Compute/Generated/Snapshot/SnapshotDeleteMethod.cs

```diff
@@ -48,7 +48,15 @@ public override void ExecuteCmdlet()
                     string resourceGroupName = this.ResourceGroupName;
                     string snapshotName = this.SnapshotName;
 
-                    var result = SnapshotsClient.DeleteWithHttpMessagesAsync(resourceGroupName, snapshotName).GetAwaiter().GetResult();
+                    Rest.Azure.AzureOperationResponse result;
+                    if (NoWait.IsPresent)
+                    {
+                        result = SnapshotsClient.BeginDeleteWithHttpMessagesAsync(resourceGroupName, snapshotName).GetAwaiter().GetResult();
+                    }
+                    else
+                    {
+                        result = SnapshotsClient.DeleteWithHttpMessagesAsync(resourceGroupName, snapshotName).GetAwaiter().GetResult();
+                    }
```

Comment:
I am not sure the GetOperationIdFromUrl call will work, so let's not do this oen and discuss with compiute

Reply:
Ok. Done.